src/Sarif.Viewer.VisualStudio/ResultTextMarker.cs

```diff
@@ -57,478 +59,286 @@ public ResultTextMarker(int runId, Region region, string fullFilePath)
+                    FileRegionsCache regionsCache = CodeAnalysisResultManager.Instance.RunDataCaches[_runId].FileRegionsCache;
+                    Region = regionsCache.PopulateTextRegionProperties(Region, uri, true);
+                }
 
-                // Navigate the caret to the desired location. Text span uses 0-based indexes
-                TextSpan ts;
-                ts.iStartLine = sourceLocation.StartLine - 1;
-                ts.iEndLine = sourceLocation.EndLine - 1;
-                ts.iStartIndex = Math.Max(sourceLocation.StartColumn - 1, 0);
-                ts.iEndIndex = Math.Max(sourceLocation.EndColumn - 1, 0);
+                IVsWindowFrame windowFrame = SdkUIUtilities.OpenDocument(ServiceProvider.GlobalProvider, this.FullFilePath, usePreviewPane);
+                if (windowFrame == null)
+                {
+                    return false;
+                }
 
-                textView.EnsureSpanVisible(ts);
-                textView.SetSelection(ts.iStartLine, ts.iStartIndex, ts.iEndLine, ts.iEndIndex);
-            }
-            return windowFrame;
-        }
+                // After the document has been opened, then the tagging should have already occurred.
+                // So let's use try to navigate again.
+                if (this.TryNavigateTo(usePreviewPane, retryNaviation: false))
+                {
+                    return true;
+                }
 
-        /// <summary>
-        /// Get source location of current marker (tracking code place). 
-        /// </summary>
-        /// <returns>
-        /// This is clone of stored source location with actual source code coordinates.
-        /// </returns>
-        public Region GetSourceLocation()
-        {
-            Region sourceLocation = Region.DeepClone();                
-            SaveCurrentTrackingData(sourceLocation);
-            return sourceLocation;
-        }
+                // Alright, navigating again didn't work, so now let's just select
+                // the region.
+                if (!TryCreateTextSpanWithinDocumentFromSourceRegion(this.Region, windowFrame, out TextSpan documentSpan))
+                {
+                    return false;
+                }
 
-        /// <summary>
-        /// Clear all markers and tracking classes
-        /// </summary>
-        public void Clear()
-        {
-            if (_marker != null)
-            {
-                RemoveHighlightMarker();
-            }
+                if (!SdkUIUtilities.TryGetTextViewFromFrame(windowFrame, out IVsTextView vsTextView))
+                {
+                    return false;
+                }
 
-            if (IsTracking())
-            {
-                RemoveTracking();
+                vsTextView.EnsureSpanVisible(documentSpan);
+                vsTextView.SetSelection(documentSpan.iStartLine, documentSpan.iStartIndex, documentSpan.iEndLine, documentSpan.iEndIndex);
             }
 
-            _tagger = null;
+            return true;
```

Comment:
>true [](start = 19, length = 4)

seems like this should be false, if nothing was done, but I might not be understanding the full context here

Reply:
Looking...

---
In reply to: [481461486](https://github.com/microsoft/sarif-visualstudio-extension/pull/191#discussion_r481461486) [](ancestors = 481461486)

Reply:
You are correct. Fixing this.

---
In reply to: [481462918](https://github.com/microsoft/sarif-visualstudio-extension/pull/191#discussion_r481462918) [](ancestors = 481462918,481461486)